### ISSUE TITLE
[Go] Implement new Concept Exercise: booleans

### ISSUE THREAD

**Issue by oanaOM** [exercism/v3#1626]
This issue describes how to implement the booleans concept exercise for the Go track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the basics of the Concept of Booleans in Go.

## Learning objectives

* Know of the existence of the bool type and its two values.
* Know about boolean operators and how to build logical expressions with them.
* Know of the boolean operator precedence rules.

## Out of scope

* Pattern matching on booleans

## Concepts

The Concepts this exercise unlocks are:
booleans-basic: know of the existence of the bool type and its two values; know about boolean operators and how to build logical expressions with them; know of the boolean operator precedence rules.

## Prerequisites

There are no prerequisites

### Hints

* Boolean operators[https://golang.org/ref/spec#Boolean_types]

### After
* Boolean operators[https://golang.org/ref/spec#Boolean_types]

## Representer

This exercise does not require any specific representation logic to be added to the representer.

## Analyzer

This exercise does not require any specific logic to be added to the analyzer.

## Implementing

To implement the concept exercise, the following files must be created:
. 
|---- .docs 
|.   |---- after.md
|.   |---- hints.md 
|.   |---- instructions.md
 |.   |---- introduction.md 
|---- .meta 
|.    |---- config.json
 |.    |---- design.md 
|.    |---- Example.go
 |---- booleans.go
 |---- booleans_test.go

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by oanaOM** [comment]
I would love to work on this. This is a my first open-source issue so I think it's a good practice for start.

**Comment by oanaOM** [comment]
@tehsphinx is it ok if I start to work on this concept exercise?

**Comment by tehsphinx** [comment]
Yes! You're most welcome to do so!

**Comment by oanaOM** [comment]
Great! I've open a new PR #1832 for this.

**Comment by tehsphinx** [comment]
Can we close this? @oanaOM?

**Comment by oanaOM** [comment]
Sure thing! closing now.
